syntax = "proto3";

service IndividualProductDiscount {
  rpc FetchDiscount (GetDiscountRequest) returns (GetDiscountResponse) {};
}

message GetDiscountRequest {
  int64 product_id = 1;
  int64 user_id = 2;
}

message GetDiscountResponse {
  float percentage = 1;
  int64 value_in_cents = 2;
}
